internal/core/src/index/StringIndexMarisa.cpp

```diff
@@ -469,7 +531,12 @@ StringIndexMarisa::Range(std::string lower_bound_value,
             std::string_view(agent.key().ptr(), agent.key().length());
         if (val > upper_bound_value ||
             (!ub_inclusive && val == upper_bound_value)) {
-            break;
+            // we could only break when trie in lexicographic order.
+            if (in_lexico_oder) {
```

Comment:
when node_order is `MARISA_WEIGHT_ORDER`, previous implementation is `break`,  but current is `continue`.
Does this logic meet expectations.

Reply:
current is still break, `MARISA_WEIGHT_ORDER` means `in_lexico_order` is false

Reply:
Using `MARISA_WEIGHT_ORDER` iteration cannot guarantee the order is lexicographic, so the `break` previously is logically incorrect, since there maybe more valid candidates after current node